x-pack/plugins/security/server/authorization/api_authorization.ts

```diff
@@ -53,15 +55,72 @@ export function initAPIAuthorization(
 
       const authz = security.authz as AuthzEnabled;
 
-      const { requestedPrivileges, requestedReservedPrivileges } = authz.requiredPrivileges.reduce(
+      const normalizeRequiredPrivileges = async (
+        privileges: AuthzEnabled['requiredPrivileges']
+      ) => {
+        const hasOperatorPrivileges = privileges.some(
+          (privilege) =>
+            privilege === ReservedPrivilegesSet.operator ||
+            (typeof privilege === 'object' &&
+              privilege.allRequired?.includes(ReservedPrivilegesSet.operator))
+        );
+
+        // nothing to normalize
+        if (!hasOperatorPrivileges) {
+          return privileges;
+        }
+
+        const esClient = await getClusterClient();
+        const operatorPrivilegesConfig = await esClient.asInternalUser.transport.request<{
```

Comment:
IMO this call is too expensive to put in a hot path such as this. I think we should look into a way to cache this during startup.

Reply:
I'd share your concerns if this was a different ES setting. Operator Privileges are only used by our orchestrated offerings, and are configured via files that reside on each node. I do not expect the same cluster to switch modes at runtime, and we explicitly do not support direct usage of this feature:

https://www.elastic.co/guide/en/elasticsearch/reference/current/operator-privileges.html
![Larry Gregory 2024-12-05 at 11 06 18](https://github.com/user-attachments/assets/21375e48-0e7f-44e1-aeb2-7721740a1b50)

Since we are in control over the configuration of Operator Privileges, it seems like it'd be sufficient to check this once on startup, and require a restart in the event the cluster somehow switched between using Operator Privileges and not using Operator Privileges.

Reply:
We make a call to operator privileges config only on startup